core/src/main/java/com/amplifyframework/storage/StorageItem.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.storage
+
+import com.amplifyframework.annotations.InternalAmplifyApi
+import java.util.Date
+
+/**
+ * Used to store the data on each item in a storage.
+ * @param path The unique path of the object in storage.
+ * @param key The unique identifier of the object in storage.
+ * @param size Size in bytes of the object
+ * @param lastModified The date the Object was Last Modified
+ * @param eTag The entity tag is an MD5 hash of the object.
+ * ETag reflects only changes to the contents of an object, not its metadata.
+ * @param pluginResults Additional results specific to the plugin.
+ */
+data class StorageItem @InternalAmplifyApi constructor(
+    /**
+     * Get unique path of the object in storage.
+     * @return Unique path of the object in storage.
+     */
+    val path: String,
```

Comment:
The idea is path is resolved to a string before storing this item right?

Reply:
A StorageItem comes from the list operation and each StorageItem is information about a file on s3. There is no StoragePath, just a string path since this is the information coming from s3 service.